### ISSUE TITLE
[C#] Setting a new parent on a node causes the components on that node to no longer be updated

### ISSUE THREAD

**Issue by mattbenic** [AtomicGameEngine/AtomicGameEngine#1341]
In the reproduction case added by branch https://github.com/LumaDigital/AtomicTests/tree/MB_AddParentNoUpdate playing the project will show an initially animating chest, which stops rotating after 10 frames. This happens when the chest's node is parented to a different node. 

The following is output to AtomicPlayer.log:
```
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update

[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Parent Update
[Tue Jan 10 15:14:59 2017] DEBUG: Child Update
[Tue Jan 10 15:14:59 2017] DEBUG: Reparent
[Tue Jan 10 15:15:04 2017] DEBUG: Parent Update
[Tue Jan 10 15:15:04 2017] DEBUG: Parent Update
[Tue Jan 10 15:15:04 2017] DEBUG: Parent Update
[Tue Jan 10 15:15:04 2017] DEBUG: Parent Update
[Tue Jan 10 15:15:04 2017] DEBUG: Parent Update
```
Note the alternating Child and Parent updates, until the reparent, after which only the parent update happens.

**Comment by mattbenic** [comment]
I started looking into this briefly, if you haven't had a chance to get to it today I'll continue. I need it for what I'm working on now, and it's fairly clearly something to do with components being removed but not added on a reparent in AtomicNET

**Comment by JoshEngebretson** [comment]
I'll give it a look into today and see if I can pick it off quick, will report :)

**Comment by mattbenic** [comment]
Thanks :)

**Comment by JoshEngebretson** [comment]
@mattbenic I went to look into this and some stuff got shaved off in the test:  https://github.com/LumaDigital/AtomicTests/blob/MB_AddParentNoUpdate/CSharpTests/SetParentNoUpdates/Resources/Components/ReparentTest.cs#L15

I would probably blame squirrels :)  Can you update with the test code you were using and I can look at it first thing tomorrow?

**Comment by mattbenic** [comment]
That's bizarre :/ I've just done a git clean and there's unfinished code in a class that was added-which makes no sense since I only committed after I ran the test. Or at least that's what I remember. I may be going senile >.>

**Comment by JoshEngebretson** [comment]
Closed via #1349 :camel: 👍
